### PR TITLE
Closes mozilla-mobile/fenix#4979 - Support Fennec web apps

### PR DESCRIPTION
**Pull request by NotWoods**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by jonalmeida** [comment]
The issue you linked doesn't exist. :)

**Comment by pocmo** [comment]
Yeah, that looks like the fenix issue: https://github.com/mozilla-mobile/fenix/issues/4979 - Let's make sure to also remove this from the commit message. :)

File: components/feature/pwa/src/main/java/mozilla/components/feature/pwa/intent/FennecWebAppIntentProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+     * A custom tab config is also set so a custom tab toolbar can be shown when the user leaves
+     * the scope defined in the manifest.
+     */
+    override suspend fun process(intent: Intent): Boolean {
+        val safeIntent = intent.toSafeIntent()
+        val url = safeIntent.dataString
+
+        return if (!url.isNullOrEmpty() && matches(intent)) {
+            val webAppManifest = fromFile(safeIntent.getStringExtra(EXTRA_FENNEC_MANIFEST_PATH))
+                ?: return false
```
Comment (pocmo) [inline components/feature/pwa/src/main/java/mozilla/components/feature/pwa/intent/FennecWebAppIntentProcessor.kt]: Do you think it would be worthwhile to save the recovered manifest in our storage and in future launches only look at the start URL but use the manifest from our storage (which may have been updated)?

File: components/service/digitalassetlinks/build.gradle [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    defaultConfig {
+        minSdkVersion config.minSdkVersion
+        targetSdkVersion config.targetSdkVersion
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
```
Comment (pocmo) [inline components/service/digitalassetlinks/build.gradle]: That file is unrelated, right?

**Comment by NotWoods** [comment]
This is currently a draft because it's untested, I still need to figure out how to sign Fennec locally for testing.

File: components/feature/pwa/src/main/java/mozilla/components/feature/pwa/intent/FennecWebAppIntentProcessor.kt (hunk 1 above)
Comment (NotWoods) [reply components/feature/pwa/src/main/java/mozilla/components/feature/pwa/intent/FennecWebAppIntentProcessor.kt]: Yes, that makes sense. I'll build on top of the manifest update code and add in functionality to "promote" the shortcut as well.

**Comment by pocmo** [comment]
> This is currently a draft because it's untested, I still need to figure out how to sign Fennec locally for testing.

Are you still stuck here? I can help you with that.

The wiki page has some hints: https://github.com/mozilla-mobile/fenix/wiki/Fennec-Migration#-development

In a nutshell you'll need to:
- Check out the esr68 branch.
- Run `./mach bootstrap` and pick Fennec artifact build. Then follow the steps.
- After that run `./mach build && ./mach package && ./mach install`. This will build, package and install Fennec on your connected device or emulator. This build will have a custom package name and will be signed with the local debug key.
- The wiki page explains how you can setup sample browser or fenix so that you can replace your local fennec build with a customized sample browser / fenix build.

Let me know if you need help along the way!

**Comment by NotWoods** [comment]
Replaced by mozilla-mobile/fenix#8159
